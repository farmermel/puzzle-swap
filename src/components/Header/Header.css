.app-title {
  margin: 0px;
  color: #260120;
  font-size: 4em; }

.App-header {
  display: flex;
  justify-content: space-between;
  position: fixed;
  top: 0;
  background-color: #fdfdfd;
  /*background-color: #D2DDDC;*/
  /*border-bottom: 1px solid #EFEDED;*/
  height: 100px;
  padding: 20px;
  color: white;
  box-shadow: 0 2px 4px #bababa;
  width: 100%;
  z-index: 1; }

.puzzleicon {
  width: 80px; }

#header-wrap {
  display: flex; }

.user-wrap {
  display: flex;
  align-items: flex-end;
  flex-direction: column; }

.user-wrap p {
  color: black;
  margin: 0 0 7px 0; }

.login {
  margin: 0 15px 0 0; }

.username-display {
  z-index: 3;
  color: black;
  position: absolute;
  right: 98px;
  font-weight: 200; }

nav {
  top: 0;
  right: 0;
  z-index: 1;
  width: 100%;
  position: absolute; }
  nav .menu {
    top: 0px;
    right: 40px;
    margin: 0;
    z-index: 1;
    width: 14%;
    background: #F6CA4F;
    padding: 3em 0 2em;
    position: absolute;
    transform: translateY(-100%);
    transition: transform 0.3s cubic-bezier(1, 0.5, 0, 1); }
    nav .menu.active {
      transform: translateY(0); }
    nav .menu li {
      list-style: none; }
      nav .menu li a {
        padding: 1em;
        color: #fff;
        display: block;
        background: none;
        line-height: 1em;
        font-size: 1.125em;
        text-align: center;
        text-decoration: none;
        transition: background 0.2s ease-in-out; }
        nav .menu li a span {
          border-bottom: 1px dotted #fff; }
        nav .menu li a:hover, nav .menu li a:focus, nav .menu li a:active {
          background: rgba(255, 255, 255, 0.2); }

.menu-toggle {
  top: 16px;
  z-index: 2;
  right: 65px;
  width: 22px;
  height: 4px;
  padding: 8px 0;
  cursor: pointer;
  border-radius: 50%;
  position: absolute; }
  .menu-toggle span {
    height: 4px;
    display: block;
    overflow: hidden;
    text-indent: 100%;
    background: #F6CA4F;
    border-radius: 2px;
    white-space: nowrap;
    transition: background 0.3s ease-in-out; }
  .menu-toggle::before, .menu-toggle::after {
    left: 0;
    content: '';
    width: 22px;
    height: 4px;
    background: #F6CA4F;
    border-radius: 2px;
    position: absolute;
    transform-origin: center center;
    transition: width 0.2s ease-in-out, background 0.2s ease-in-out, transform 0.2s ease-in-out, top 0.2s ease-in-out, bottom 0.2s ease-in-out, left 0.2s ease-in-out, right 0.2s ease-in-out; }
  .menu-toggle::before {
    top: 0; }
  .menu-toggle::after {
    bottom: 0; }
  .menu-toggle.menu-open span {
    background: rgba(255, 255, 255, 0); }
  .menu-toggle.menu-open::before {
    top: 8px;
    transform: rotate(-45deg); }
  .menu-toggle.menu-open::after {
    bottom: 8px;
    transform: rotate(45deg); }
  .menu-toggle.menu-open::before, .menu-toggle.menu-open::after {
    width: 26px;
    background: #fff; }
  .menu-toggle.menu-open:hover span, .menu-toggle.menu-open:focus span, .menu-toggle.menu-open:active span {
    background: rgba(255, 255, 255, 0); }
  .menu-toggle.menu-open:hover::before, .menu-toggle.menu-open:focus::before, .menu-toggle.menu-open:active::before {
    top: 8px;
    left: -3px;
    transform: rotate(-45deg); }
  .menu-toggle.menu-open:hover::after, .menu-toggle.menu-open:focus::after, .menu-toggle.menu-open:active::after {
    left: 9px;
    bottom: 8px;
    transform: rotate(45deg); }
  .menu-toggle.menu-open:hover::before, .menu-toggle.menu-open:hover::after, .menu-toggle.menu-open:focus::before, .menu-toggle.menu-open:focus::after, .menu-toggle.menu-open:active::before, .menu-toggle.menu-open:active::after {
    width: 18px; }
  .menu-toggle:hover span, .menu-toggle:focus span, .menu-toggle:active span {
    background: rgba(255, 255, 255, 0); }
  .menu-toggle:hover::before, .menu-toggle:focus::before, .menu-toggle:active::before {
    top: 8px;
    left: -3px;
    transform: rotate(45deg); }
  .menu-toggle:hover::after, .menu-toggle:focus::after, .menu-toggle:active::after {
    left: 9px;
    bottom: 8px;
    transform: rotate(-45deg); }
  .menu-toggle:hover::before, .menu-toggle:hover::after, .menu-toggle:focus::before, .menu-toggle:focus::after, .menu-toggle:active::before, .menu-toggle:active::after {
    width: 18px; }

@media screen and (max-width: 520px) {
  .app-title {
    padding: 14px 0 0 7px; }
  .puzzleicon {
    width: 60px; } }
