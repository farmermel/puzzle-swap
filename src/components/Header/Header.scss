.app-title {
  margin: 0px;
  color: #260120;
  font-size: 4em;
}

.App-header {
  display: flex;
  justify-content: start;
  position: fixed;
  top: 0;
  background-color: #fdfdfd;
  /*background-color: #D2DDDC;*/
  /*border-bottom: 1px solid #EFEDED;*/
  height: 100px;
  padding: 20px;
  color: white;
  box-shadow: 0 2px 4px #bababa;
  width: 100%;
  z-index: 1;
}

.puzzleicon {
  width: 80px;
}

#header-wrap {
  display: flex;
}

.user-wrap {
  display: flex;
  align-items: flex-end;
  flex-direction: column;
}

.user-wrap p {
  color: black;
  margin: 0 0 7px 0;
}

.login {
  margin: 0 15px 0 0;
}

.username-display {
  z-index: 3;
  color: black;
  position: absolute;
  right: 98px;
  font-weight: 200;
}

nav {
  top: 0;
  right: 0;
  z-index: 1;
  width: 100%;
  position: absolute;
  .menu {
    top: 0px;
    right: 40px;
    margin: 0;
    z-index: 1;
    width: 14%;
    background: #F6CA4F;
    padding: 3em 0 2em;
    position: absolute;
    transform: translateY(-100%);
    transition: transform 0.3s cubic-bezier(1,0.5,0,1);
    &.active {
      transform: translateY(0);
    }
    li {
      list-style: none;
      a {
        padding: 1em;
        color: #fff;
        display: block;
        background: none;
        line-height: 1em;
        font-size: 1.125em;
        text-align: center;
        text-decoration: none;
        transition: background 0.2s ease-in-out;
        span {
          border-bottom: 1px dotted #fff;
        }
        &:hover, &:focus, &:active {
          background: rgba(255,255,255,0.2);
        }
      }
    }
  }
}

.menu-toggle {
  top: 16px;
  z-index: 2;
  right: 65px;
  width: 22px;
  height: 4px;
  padding: 8px 0;
  cursor: pointer;
  border-radius: 50%;
  position: absolute;
  span {
    height: 4px;
    display: block;
    overflow: hidden;
    text-indent: 100%;
    background: #F6CA4F;
    border-radius: 2px;
    white-space: nowrap;
    transition: background 0.3s ease-in-out;
  }
  &::before, &::after {
    left: 0;
    content: '';
    width: 22px;
    height: 4px;
    background: #F6CA4F;
    border-radius: 2px;
    position: absolute;
    transform-origin: center center;
    transition: width 0.2s ease-in-out, background 0.2s ease-in-out, transform 0.2s ease-in-out, top 0.2s ease-in-out, bottom 0.2s ease-in-out, left 0.2s ease-in-out, right 0.2s ease-in-out;
  }
  &::before {
    top: 0;
  }
  &::after {
    bottom: 0;
  }
  &.menu-open {
    span {
      background: rgba(255,255,255,0);
    }
    &::before {
      top: 8px;
      transform: rotate(-45deg);
    }
    &::after {
      bottom: 8px;
      transform: rotate(45deg);
    }
    &::before, &::after {
      width: 26px;
      background: #fff;
    }
    &:hover, &:focus, &:active {
      span {
        background: rgba(255,255,255,0);
      }
      &::before {
        top: 8px;
        left: -3px;
        transform: rotate(-45deg);
      }
      &::after {
        left: 9px;
        bottom: 8px;
        transform: rotate(45deg)
      }
      &::before, &::after {
        width: 18px;
      }
    }
  }
  &:hover, &:focus, &:active {
    span {
      background: rgba(255,255,255,0);
    }
    &::before {
      top: 8px;
      left: -3px;
      transform: rotate(45deg);
    }
    &::after {
      left: 9px;
      bottom: 8px;
      transform: rotate(-45deg)
    }
    &::before, &::after {
      width: 18px;
    }
  }
}

@media screen and (max-width: 520px) {
  .app-title {
    padding: 14px 0 0 7px;
  }

  .puzzleicon {
    width: 60px;
  }
}