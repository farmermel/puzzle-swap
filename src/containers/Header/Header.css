.app-title {
  margin: 0px;
  color: #260120;
  font-size: 4em;
}

.App-header {
  display: flex;
  justify-content: space-between;
  position: fixed;
  width: 100%;
  background-color: #fdfdfd;
  /*background-color: #D2DDDC;*/
  /*border-bottom: 1px solid #EFEDED;*/
  height: 100px;
  padding: 20px;
  color: white;
  box-shadow: 0 2px 4px #bababa;
}

.header-right {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-end;
  padding: 0 35px 0 0;
}

.header-right a {
  text-decoration: underline;
  color: black;
  height: fit-content;
  cursor: pointer;
}

.puzzleicon {
  width: 80px;
}

#header-wrap {
  display: flex;
}

.user-wrap {
  display: flex;
  align-items: flex-end;
  flex-direction: column;
}

.user-wrap p {
  color: black;
  margin: 0 0 7px 0;
}

.login {
  margin: 0 15px 0 0;
}