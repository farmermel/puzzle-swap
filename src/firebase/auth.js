import { auth } from './firebase';

export const doCreateUserWithEmailAndPassword = (email, password) =>
  auth.createUserWithEmailAndPassword(email, password);

export const doSignInWithEmailAndPassword = (email, password) => 
  auth.signInWithEmailAndPassword(email, password);

export const onAuthStateChanged = user => 
  auth.onAuthStateChanged(user);

export const doSignOut = () => 
  auth.signOut();