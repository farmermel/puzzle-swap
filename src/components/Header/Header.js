import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { auth } from '../../firebase';
import { setLogin } from '../../actions/loginAction';
import './Header.css';

export const renderNotLoggedIn = () => {
  return (
    <div className='header-right'>
      <div className='login-container'>
        <Link to={{ pathname: '/login' }}>
          <button className='login'>Login</button>
        </Link>
        <Link to={{ pathname: '/sign-up' }}>
          <button className='login'>Sign up</button>
        </Link>   
      </div>
      <Link to={{ pathname: '/messages' }}>
        <button className='messages'>Messages</button>
      </Link>
    </div>
  )
}

export const handleSignOut = async () => {
  await auth.doSignOut();
  setLogin(false);
}

export const renderLoggedIn = () => {
  return <button onClick={handleSignOut}>Sign out</button>
}

export const determineLoginButtons = (loggedIn) => {
  const display = loggedIn ? renderLoggedIn() : renderNotLoggedIn();
  return display;
}

export const Header = ({ loggedIn }) => {
  return (
    <div className="App-header">
      <Link to={{ pathname: '/' }}>
        <h1 className='app-title'>Puzzle Swap</h1>
      </Link>
          { determineLoginButtons(loggedIn) }
    </div>
  )
}

export const mapDispatchToProps = dispatch => ({
  setLogin: boolean => dispatch(setLogin(boolean))
})

export default connect(null, mapDispatchToProps)(Header);