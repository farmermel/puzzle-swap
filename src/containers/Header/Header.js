import React from 'react';
import { Link } from 'react-router-dom';
import { auth } from '../../firebase';
import puzzleIcon from '../../assets/puzzle.svg';
import PropTypes from 'prop-types';
import './Header.css';

export const renderNotLoggedIn = () => {
  return (
    <div className='header-right'>
      <div className='login-container'>
        <Link to={{ pathname: '/login' }}>
          <span className='login'>Login</span>
        </Link>
        <Link to={{ pathname: '/sign-up' }}>
          <span>Sign up</span>
        </Link>   
      </div>
      
    </div>
  )
}

export const handleSignOut = async () => {
  await auth.doSignOut();
}

export const renderLoggedIn = (user) => {
  return (
    <div className='header-right'>
      <div className='user-wrap'>
        <p>{user.username}</p>
        <a onClick={handleSignOut}>Sign out</a>
      </div>
      <Link to={{ pathname: '/messages' }}>
        <button className='messages'>Messages</button>
      </Link>
    </div>
  )
}

export const determineLoginButtons = (user) => {
  const display = user ? renderLoggedIn(user) : renderNotLoggedIn();
  return display;
}

export const Header = ({ user }) => {
  return (
    <div className="App-header">
      <Link to={{ pathname: '/' }} id='header-wrap'>
        <img src={puzzleIcon} alt='puzzle' className='puzzleicon' />
        <h1 className='app-title'>Puzzle Swap</h1>
      </Link>
          { determineLoginButtons(user) }
    </div>
  )
}

Header.propTypes = {
  user: PropTypes.objectOf(PropTypes.string)
}

export default Header;