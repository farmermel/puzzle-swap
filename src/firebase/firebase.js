import * as firebase from 'firebase';
import { firebaseKey,
  fbAuthDomain,
  fbdbUrl,
  fbProjId,
  fbMessagingSenderId } from '../helpers/apiKeys';

const config = {
  apiKey: firebaseKey,
  authDomain: fbAuthDomain,
  databaseURL: fbdbUrl,
  projectId: fbProjId,
  storageBucket: "",
  messagingSenderId: fbMessagingSenderId
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const auth = firebase.auth();

const db = firebase.database();

export {
  db,
  auth,
};