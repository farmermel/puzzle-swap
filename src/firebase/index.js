import * as auth from './auth';
import * as db from './db';
import * as firebase from './firebase';
import * as store from './store';

export {
  auth,
  db,
  store,
  firebase,
};