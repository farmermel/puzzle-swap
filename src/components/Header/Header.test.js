import React from 'react';
import { shallow } from 'enzyme';
import { auth } from '../../firebase';
import { Header } from './Header';

describe('Header', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Header user={{username: 'case', uid: '5'}}
                              hasErrored={jest.fn()} />);
  })

  it('should match snapshot when user is logged in', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot when user is not logged in', () => {
    wrapper = shallow(<Header user={null}
                              hasErrored={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  })

  describe('handleSignOut', () => {
    it('calls doSignOut method on auth', () => {
      auth.doSignOut = jest.fn();
      expect(auth.doSignOut).not.toHaveBeenCalled();
      wrapper.instance().handleSignOut();
      expect(auth.doSignOut).toHaveBeenCalled();
    })

    it('catches error and calls hasErrored error message if anything errors', async () => {
      auth.doSignOut = jest.fn().mockImplementation(() => {
        throw new Error('failed to sign out')
      }); 
      await wrapper.instance().handleSignOut();
      expect(wrapper.instance().props.hasErrored).toHaveBeenCalledWith('failed to sign out');
    })
  })
})