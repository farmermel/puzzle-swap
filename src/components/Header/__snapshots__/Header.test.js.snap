// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot when user is logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    loggedIn={true}
    setLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to={
                    Object {
                              "pathname": "/",
                            }
          }
>
          <h1
                    className="app-title"
          >
                    Puzzle Swap
          </h1>
</Link>,
        <button
          onClick={[Function]}
>
          Sign out
</button>,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1
            className="app-title"
>
            Puzzle Swap
</h1>,
          "replace": false,
          "to": Object {
            "pathname": "/",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Puzzle Swap",
            "className": "app-title",
          },
          "ref": null,
          "rendered": "Puzzle Swap",
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign out",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Sign out",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to={
                        Object {
                                    "pathname": "/",
                                  }
            }
>
            <h1
                        className="app-title"
            >
                        Puzzle Swap
            </h1>
</Link>,
          <button
            onClick={[Function]}
>
            Sign out
</button>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h1
              className="app-title"
>
              Puzzle Swap
</h1>,
            "replace": false,
            "to": Object {
              "pathname": "/",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Puzzle Swap",
              "className": "app-title",
            },
            "ref": null,
            "rendered": "Puzzle Swap",
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign out",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Sign out",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header should match snapshot when user is not logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    loggedIn={false}
    setLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to={
                    Object {
                              "pathname": "/",
                            }
          }
>
          <h1
                    className="app-title"
          >
                    Puzzle Swap
          </h1>
</Link>,
        <div
          className="header-right"
>
          <div
                    className="login-container"
          >
                    <Link
                              replace={false}
                              to={
                                        Object {
                                                  "pathname": "/login",
                                                }
                              }
                    >
                              <button
                                        className="login"
                              >
                                        Login
                              </button>
                    </Link>
                    <Link
                              replace={false}
                              to={
                                        Object {
                                                  "pathname": "/sign-up",
                                                }
                              }
                    >
                              <button
                                        className="login"
                              >
                                        Sign up
                              </button>
                    </Link>
          </div>
          <Link
                    replace={false}
                    to={
                              Object {
                                        "pathname": "/messages",
                                      }
                    }
          >
                    <button
                              className="messages"
                    >
                              Messages
                    </button>
          </Link>
</div>,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1
            className="app-title"
>
            Puzzle Swap
</h1>,
          "replace": false,
          "to": Object {
            "pathname": "/",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Puzzle Swap",
            "className": "app-title",
          },
          "ref": null,
          "rendered": "Puzzle Swap",
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="login-container"
>
              <Link
                            replace={false}
                            to={
                                          Object {
                                                        "pathname": "/login",
                                                      }
                            }
              >
                            <button
                                          className="login"
                            >
                                          Login
                            </button>
              </Link>
              <Link
                            replace={false}
                            to={
                                          Object {
                                                        "pathname": "/sign-up",
                                                      }
                            }
              >
                            <button
                                          className="login"
                            >
                                          Sign up
                            </button>
              </Link>
</div>,
            <Link
              replace={false}
              to={
                            Object {
                                          "pathname": "/messages",
                                        }
              }
>
              <button
                            className="messages"
              >
                            Messages
              </button>
</Link>,
          ],
          "className": "header-right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to={
                                    Object {
                                                      "pathname": "/login",
                                                    }
                  }
>
                  <button
                                    className="login"
                  >
                                    Login
                  </button>
</Link>,
                <Link
                  replace={false}
                  to={
                                    Object {
                                                      "pathname": "/sign-up",
                                                    }
                  }
>
                  <button
                                    className="login"
                  >
                                    Sign up
                  </button>
</Link>,
              ],
              "className": "login-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="login"
>
                    Login
</button>,
                  "replace": false,
                  "to": Object {
                    "pathname": "/login",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                    "className": "login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "button",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="login"
>
                    Sign up
</button>,
                  "replace": false,
                  "to": Object {
                    "pathname": "/sign-up",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign up",
                    "className": "login",
                  },
                  "ref": null,
                  "rendered": "Sign up",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="messages"
>
                Messages
</button>,
              "replace": false,
              "to": Object {
                "pathname": "/messages",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Messages",
                "className": "messages",
              },
              "ref": null,
              "rendered": "Messages",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to={
                        Object {
                                    "pathname": "/",
                                  }
            }
>
            <h1
                        className="app-title"
            >
                        Puzzle Swap
            </h1>
</Link>,
          <div
            className="header-right"
>
            <div
                        className="login-container"
            >
                        <Link
                                    replace={false}
                                    to={
                                                Object {
                                                            "pathname": "/login",
                                                          }
                                    }
                        >
                                    <button
                                                className="login"
                                    >
                                                Login
                                    </button>
                        </Link>
                        <Link
                                    replace={false}
                                    to={
                                                Object {
                                                            "pathname": "/sign-up",
                                                          }
                                    }
                        >
                                    <button
                                                className="login"
                                    >
                                                Sign up
                                    </button>
                        </Link>
            </div>
            <Link
                        replace={false}
                        to={
                                    Object {
                                                "pathname": "/messages",
                                              }
                        }
            >
                        <button
                                    className="messages"
                        >
                                    Messages
                        </button>
            </Link>
</div>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h1
              className="app-title"
>
              Puzzle Swap
</h1>,
            "replace": false,
            "to": Object {
              "pathname": "/",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Puzzle Swap",
              "className": "app-title",
            },
            "ref": null,
            "rendered": "Puzzle Swap",
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="login-container"
>
                <Link
                                replace={false}
                                to={
                                                Object {
                                                                "pathname": "/login",
                                                              }
                                }
                >
                                <button
                                                className="login"
                                >
                                                Login
                                </button>
                </Link>
                <Link
                                replace={false}
                                to={
                                                Object {
                                                                "pathname": "/sign-up",
                                                              }
                                }
                >
                                <button
                                                className="login"
                                >
                                                Sign up
                                </button>
                </Link>
</div>,
              <Link
                replace={false}
                to={
                                Object {
                                                "pathname": "/messages",
                                              }
                }
>
                <button
                                className="messages"
                >
                                Messages
                </button>
</Link>,
            ],
            "className": "header-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to={
                                        Object {
                                                            "pathname": "/login",
                                                          }
                    }
>
                    <button
                                        className="login"
                    >
                                        Login
                    </button>
</Link>,
                  <Link
                    replace={false}
                    to={
                                        Object {
                                                            "pathname": "/sign-up",
                                                          }
                    }
>
                    <button
                                        className="login"
                    >
                                        Sign up
                    </button>
</Link>,
                ],
                "className": "login-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <button
                      className="login"
>
                      Login
</button>,
                    "replace": false,
                    "to": Object {
                      "pathname": "/login",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Login",
                      "className": "login",
                    },
                    "ref": null,
                    "rendered": "Login",
                    "type": "button",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <button
                      className="login"
>
                      Sign up
</button>,
                    "replace": false,
                    "to": Object {
                      "pathname": "/sign-up",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sign up",
                      "className": "login",
                    },
                    "ref": null,
                    "rendered": "Sign up",
                    "type": "button",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="messages"
>
                  Messages
</button>,
                "replace": false,
                "to": Object {
                  "pathname": "/messages",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Messages",
                  "className": "messages",
                },
                "ref": null,
                "rendered": "Messages",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
